src/nspawn/nspawn-util.c

```diff
@@ -58,11 +59,17 @@ int systemd_installation_has_version(const char *root, unsigned minimal_version)
                         if (!t2)
                                 continue;
 
-                        t2[0] = '\0'; /* truncate the suffix */
+                        /* truncate the version suffix at the first dash, underscore, or dot. */
+                        t2 = t + strcspn(t, "-_.");
 
-                        r = safe_atou(t, &version);
+                        string = strndup(t, t2 - t);
```

Comment:
hmm, why do you calculate `t2` as pointer first, just to then get back to a size right after, i.e. just change this line to:

```c
string = strndup(t, strcspn(t, "-_."));
```
and don#t bother with t2